operations/host_spawn.go

```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
 	}
 }
 
+func hostSSH() cli.Command {
+	const (
+		identityFlagName = "identity"
+	)
+
+	return cli.Command{
+		Name:  "ssh",
+		Usage: "ssh into a spawn host",
+		Flags: addHostFlag(
+			cli.StringFlag{
+				Name:  joinFlagNames(identityFlagName, "i"),
+				Usage: "Path to a specific identity (private key), for ssh -i",
+			},
+		),
+		Before: mergeBeforeFuncs(setPlainLogger, requireHostFlag),
+		Action: func(c *cli.Context) error {
+			confPath := c.Parent().Parent().String(confFlagName)
+			hostID := c.String(hostFlagName)
+			key := c.String(identityFlagName)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			conf, err := NewClientSettings(confPath)
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			client := conf.setupRestCommunicator(ctx)
+			defer client.Close()
+
+			h, err := client.GetSpawnHost(ctx, hostID)
+			if err != nil {
+				return errors.Wrap(err, "problem getting host")
+			}
+			if h == nil {
+				return errors.New("host not found")
+			}
+			if restModel.FromStringPtr(h.Status) != evergreen.HostRunning {
+				return errors.New("host is not running")
+			}
+			if restModel.FromStringPtr(h.User) == "" || restModel.FromStringPtr(h.HostURL) == "" {
+				return errors.New("unable to ssh into host without user or DNS name")
+			}
+			args := []string{"ssh", "-tt", fmt.Sprintf("%s@%s", *h.User, *h.HostURL)}
```

Comment:
should these (`h.User` and `h.HostURL`) use FromStringPtr as well?

Reply:
I can change it, although I'd argue it's not necessary since  we already know they aren't nil.

Reply:
agreed